km/km_filesys.c

```diff
@@ -2986,14 +3083,21 @@ static inline int km_fs_recover_socket(km_nt_socket_t* nt_sock, struct sockaddr*
       }
 
       if (nt_sock->state == KM_SOCK_STATE_BIND || nt_sock->state == KM_SOCK_STATE_LISTEN) {
-         // If snapshot is being recovered right after it was taken there is a chance there are
-         // sockets in TIME_WAIT state from the remaining from the initial run, so the following
-         // bind() will fail. This avoid the falure.
-         int flag = 1;
-         if (setsockopt(host_fd, SOL_SOCKET, SO_REUSEADDR, &flag, sizeof(flag)) != 0) {
-            km_warn("recover setsockopt(SO_REUSEADDR) failed");
-            return -1;
+         // Recover the SO_REUSEADDR setting.
```

Comment:
I'm am bit turn apart here. I understand your logic here, OTOH I'm thinking about FaaS workflow where this condition is basically forced